### PR TITLE
(CDAP-16057) Allow customizing service account for Dataproc

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

**Review by albertshau — APPROVED** [review]

File: cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java [hunk 1]
```diff
@@ -341,6 +341,11 @@ public ClusterOperationMetadata createCluster(String name, String imageVersion,
       GceClusterConfig.Builder clusterConfig = GceClusterConfig.newBuilder()
         .addServiceAccountScopes(DataprocConf.CLOUD_PLATFORM_SCOPE)
         .putAllMetadata(metadata);
+
+      if (conf.getServiceAccount() != null) {
```
Comment (CuriousVini) [inline cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java:345]: should we also check for empty string?

**Review by CuriousVini — APPROVED** [review]
one question, rest LGTM

File: cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java (hunk 1 above)
Comment (chtyim) [reply cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java:345]: It is already taken care of by the `DataprocConf.getString` method.

**Comment by chtyim** [comment]
Build passed https://builds.cask.co/browse/CDAP-RUT1715

**Pull request merged**